core/iwasm/libraries/thread-mgr/thread_manager.c

```diff
@@ -981,8 +1006,6 @@ wasm_cluster_cancel_thread(WASMExecEnv *exec_env)
         return 0;
     }
 
-    os_mutex_lock(&exec_env->cluster->lock);
```

Comment:
Had better add comment for this function to ask the caller to ensure thread safety

Reply:
Removing the code here since clusters_have_exec_env will lock again, which leads to hang.